attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -37,20 +43,38 @@ impl Attester for TdxAttester {
 
         report_data.resize(64, 0);
 
-        let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
-            d: report_data.as_slice().try_into()?,
-        };
-
-        let engine = base64::engine::general_purpose::STANDARD;
-        let quote = match tdx_attest_rs::tdx_att_get_quote(Some(&tdx_report_data), None, None, 0) {
-            (tdx_attest_rs::tdx_attest_error_t::TDX_ATTEST_SUCCESS, Some(q)) => engine.encode(q),
-            (error_code, _) => {
-                return Err(anyhow!(
-                    "TDX Attester: Failed to get TD quote. Error code: {:?}",
-                    error_code
-                ));
+        let quote_bytes = match TsmReportPath::open() {
+            Ok(tsm) => match tsm.attestation_report(TsmReportProvider::Tdx(report_data)) {
+                Ok(bytes) => {
+                    tsm.close();
+                    bytes
+                }
+                Err(e) => {
+                    tsm.close();
+                    return Err(anyhow!("TDX Attester: {}", e));
+                }
+            },
+            #[cfg(feature = "tdx-deprecated-getquote-ioctl")]
+            Err(_) => {
+                let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
+                    d: report_data.as_slice().try_into().unwrap(),
```

Comment:
```suggestion
                    d: report_data.as_slice().try_into().ok_or_else(|| anyhow!(..))?,
```

Reply:
I don't have a way to test the deprecated flows so we could leave the change to another PR since it's not related to this PR

Reply:
sorry, I took a closer look and looks like there's a minor change in it. I'm changing it back.